lib/ansible/modules/cloud/docker/docker_node.py

```diff
@@ -0,0 +1,290 @@
+            - Swarm name of the node.
+            - Used with I(state=update).
+    labels:
+        description: User-defined key/value metadata.
+    labels_state:
+        description:
+            - Defines the operation on the lables assigned to node.
+            - Set to C(merge) to merge provided labels with already assigned ones. It will update existing keys.
+              The I(labels) must be specified.
+            - Set to C(replace) to replace assigned labels with provided ones. The I(labels) must be specified.
+            - Set to C(remove) to remove all labels assigned to node.
+        choices:
+          - merge
+          - replace
+          - remove
+        default: 'merge'
+    availability:
+        description: Node availability status to assign
+        choices:
+          - active
+          - pause
+          - drain
+    role:
+        description: Node role to assign
+        choices:
+          - manager
+          - worker
+extends_documentation_fragment:
+    - docker
+requirements:
+    - "python >= 2.6"
+    - "docker-py >= 1.10.0"
+    - "Please note that the L(docker-py,https://pypi.org/project/docker-py/) Python
+       module has been superseded by L(docker,https://pypi.org/project/docker/)
+       (see L(here,https://github.com/docker/docker-py/issues/1310) for details).
+       For Python 2.6, C(docker-py) must be used. Otherwise, it is recommended to
+       install the C(docker) Python module. Note that both modules should I(not)
+       be installed at the same time. Also note that when both modules are installed
+       and one of them is uninstalled, the other might no longer function and a
+       reinstall of it is required."
+    - "The docker server >= 1.10.0"
+author:
+  - Piotr Wojciechowski (@wojciechowskipiotr)
+  - Thierry Bouvet (@tbouvet)
+
+'''
+
+EXAMPLES = '''
+- name: List names of registered swarm nodes
+  docker_node:
+    state: list
+
+- name: Set node role
+  docker_node:
+    state: update
+    name: mynode
+    role: manager
+
+- name: Set node availability
+  docker_node:
+    state: update
+    name: mynode
+    availability: drain
+
+- name: Replace node labels with new labels
+  docker_node:
+    state: update
+    name: mynode
+    labels:
+      key: value
+    labels_state: replace
+
+- name: Merge node labels and new labels
+  docker_node:
+    state: update
+    name: mynode
+    labels:
+      key: value
+
+- name: Merge node labels and new labels
+  docker_node:
+    state: update
+    name: mynode
+    labels:
+      key: value
+    labels_state: merge
+
+- name: Remove node labels
+  docker_node:
+    state: update
+    name: mynode
+    labels_state: remove
+'''
+
+RETURN = '''
+node_facts:
+  description: Information about node after 'update' operation
+  returned: success
+  type: dict
+actions:
+  description: Provides the actions done on the swarm.
+  returned: when action failed.
+  type: list
+  example: "['This cluster is already a swarm cluster']"
```

Comment:
Is someone actually using this (except for debugging)? It might be more useful to add proper `diff` support.

Reply:
I left it to preserve the compatibility with outputs in `docker_swarm` module. @tbouvet ?

Reply:
@tbouvet - any thoughts on that? Would also appreciate a full review as for this PR there are not many doubts to close I think.